llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -1358,3 +1358,205 @@ void llvm::applyUpdToConcat(MachineInstr &MI, MachineRegisterInfo &MRI,
 
   MI.eraseFromParent();
 }
+
+bool llvm::matchLoadStoreSplit(GLoadStore &MI, MachineRegisterInfo &MRI,
+                               const AIEBaseInstrInfo &TII,
+                               unsigned &MaxMemSize) {
+
+  Register ValReg = MI.getReg(0);
+  const LLT ValTy = MRI.getType(ValReg);
+  const bool IsLoad = isa<GLoad>(MI);
+  MaxMemSize = TII.getMaxLoadStoreSize();
```

Comment:
Is there a specific reason we are passing & setting the `MaxMemSize` in match & using that in apply ? We could have used `TII.getMaxLoadStoreSize();` directly in apply ?

Reply:
It could be. What I had in mind was a target specific match and a completely target independent apply. It gives the freedom to apply different splitting tragedies for future targets, considering different selection combines opportunities.